Model/Order/PlaceOrder.php

```diff
@@ -137,11 +181,234 @@ public function place(string $shopId, OrderDataInterface $order): ResultInterfac
+                [
+                    'errors' => [
+                        $this->errorFactory->create(
+                            [
+                                'message' => __(
+                                    'There are no successful transactions available for order "%1"',
+                                    $publicOrderId
+                                ),
+                                'code' => 500,
+                                'type' => 'server.bold_checkout_api_error'
+                            ]
+                        )
+                    ]
+                ]
+            );
+        }
+
+        /**
+         * @var array{
+         *     gateway: string,
+         *     payment_id: string,
+         *     amount: int,
+         *     transaction_id: string,
+         *     currency: string,
+         *     step: string,
+         *     status: 'success'|'',
+         *     tender_type: string,
+         *     tender_details: array{
+         *         brand: string,
+         *         last_four: string,
+         *         bin: string,
+         *         expiration: string
+         *     },
+         *     gateway_response_data: string[]
+         * } $firstTransaction
+         */
+        $firstTransaction = array_shift($transactions);
+        /** @var OrderPaymentInterface $orderPayment */
+        $orderPayment = $this->paymentFactory->create();
+        /** @var TransactionInterface $transaction */
+        $transaction = $this->transactionFactory->create();
+        /** @var OrderDataInterface $orderData */
+        $orderData = $this->orderDataFactory->create();
+        [$cardExpirationMonth, $cardExpirationYear] = explode(
+            '/',
+            $firstTransaction['tender_details']['expiration'],
+            2
+        );
+
+        $orderPayment->setBaseAmountPaid($firstTransaction['amount'] / 100);
+        $orderPayment->setAmountPaid($firstTransaction['amount'] / 100);
+        $orderPayment->setCcLast4($firstTransaction['tender_details']['last_four']);
+        $orderPayment->setCcType($firstTransaction['tender_details']['brand']);
+        $orderPayment->setCcExpMonth($cardExpirationMonth);
+        $orderPayment->setCcExpYear($cardExpirationYear);
+
+        $transaction->setTxnId($firstTransaction['transaction_id']);
+        $transaction->setTxnType(TransactionInterface::TYPE_PAYMENT); // TODO: verify this transaction type is correct
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $transaction->setAdditionalInformation('gateway', $firstTransaction['gateway']);
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $transaction->setAdditionalInformation('payment_id', $firstTransaction['payment_id']);
+        $transaction->setIsClosed(1);
+
+        $orderData->setQuoteId((int)$quoteId);
+        $orderData->setPublicId($publicOrderId);
+        $orderData->setPayment($orderPayment);
+        $orderData->setTransaction($transaction);
```

Comment:
[Clarification] Is this where we'd also add customer data?

Reply:
No, I'm just manually constructing the same object that we passed to the previous `place()` endpoint above.